### PR TITLE
HTS-287: Report missing details back to the user

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/views/register/missing_user_info.scala.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+@*
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/views/register/missing_user_info.scala.html]: this page is a WIP -  under construction

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala [hunk 2]
```diff
@@ -76,21 +75,30 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
                   Logger.error(s"Could not perform eligibility check: $error")
                   InternalServerError("")
                 }, _.result.fold(
-                  SeeOther(routes.RegisterController.notEligible().url))(
-                  userDetails ⇒ Ok(views.html.register.confirm_details(userDetails)))
+                  missingInfos ⇒ {
+                    Logger.error(s"user ${maybeNino.get} has missing information: ${missingInfos.missingInfo.mkString(",")}")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: `.get` will throw if the `Option` is `None` - it's safer to `yield` the nino from the for-comprehension above and bring it into scope here

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala [hunk 3]
```diff
@@ -157,20 +165,19 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
...
-        errors ⇒ s"User info did not pass NS&I validity checks: ${errors.toList.mkString("; ")}"
-      )
-    }
-
-    // use sequence to push the option into the either
-    val result: Either[String, Option[NSIUserInfo]] = conversion.sequence[EitherOrString,NSIUserInfo]
+    val mayBeNSIUserInfo: Either[String, Option[NSIUserInfo]] = eligibilityResult.result.fold(
+      _ ⇒ Right(None), {
+        case Some(info) ⇒
+          NSIUserInfo(info).toEither.leftMap(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: can also use `bimap` to avoid having to do `leftMap` and `map` separately

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**